### PR TITLE
#19424 A regex pattern is used to extract the task id to support calver notation

### PR DESCRIPTION
**Pull request by nollymar**
The substring to extract the task id was replaced by a regex expression to support several lengths.

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(DataProviderRunner.class)
+public class StartupTasksExecutorTest {
+
+    @DataProvider
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java:16]: Codacy found an issue: [JUnit 4 tests that execute tests should use the @Test annotation, JUnit 5 tests should use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java [hunk 2]
```diff
@@ -169,7 +185,7 @@ public void executeUpgrades() throws DotDataException {
         for (Class<?> c : TaskLocatorUtil.getStartupRunOnceTaskClasses()) {
             name = c.getCanonicalName();
             name = name.substring(name.lastIndexOf(".") + 1);
-            String id = name.substring(4, 9);
+            String id = getTaskId(name);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java:188]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        String taskName;
+
+        public TestCase(final String taskId, final String taskName){
+            this.taskId = taskId;
+            this.taskName = taskName;
+        }
+    }
+
+    @BeforeClass
+    public static void prepare() throws Exception{
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java:36]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    /**
+     * Method to test: {@link StartupTasksExecutor#getTaskId(String)}
+     * Test case: A task name whose id has a specific length
+     * Expected result: It should extract the length correctly
+     * @param testCase
+     */
+    @Test
+    @UseDataProvider("testCases")
+    public void testGetTaskId(TestCase testCase){
+        final String id = StartupTasksExecutor.getInstance().getTaskId(testCase.taskName);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java:50]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java [hunk 5]
```diff
@@ -149,9 +152,22 @@ public void executeStartUpTasks() throws DotDataException {
...
-    
+
+    /**
+     * Returns the id part of a task name
+     * @param taskName
+     * @return
+     */
+    @VisibleForTesting
+    String getTaskId(final String taskName){
+        final Pattern pattern = Pattern.compile("[0-9]+");
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java:163]: it is usually a good idea to have the pattern Compile declared static as it is immutable. and therefore thread-safe.
The matcher has to be recreated every time.  So I would  move final Pattern pattern = Pattern.compile("[0-9]+"); to a static constant.

**Review by fabrizzio-dotCMS — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java:16]: Codacy found an issue: [JUnit 4 tests that execute tests should use the @Test annotation, JUnit 5 tests should use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java:36]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java:50]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java [hunk 6]
```diff
@@ -169,7 +184,7 @@ public void executeUpgrades() throws DotDataException {
         for (Class<?> c : TaskLocatorUtil.getStartupRunOnceTaskClasses()) {
             name = c.getCanonicalName();
             name = name.substring(name.lastIndexOf(".") + 1);
-            String id = name.substring(4, 9);
+            String id = getTaskId(name);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java:187]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java:16]: Codacy found an issue: [JUnit 4 tests that execute tests should use the @Test annotation, JUnit 5 tests should use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java:36]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java:50]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java:187]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java:187]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java:16]: Codacy found an issue: [JUnit 4 tests that execute tests should use the @Test annotation, JUnit 5 tests should use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java:36]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/StartupTasksExecutorTest.java:50]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354433)

**Pull request merged**